app/controllers/synapses_controller.rb

```diff
@@ -22,10 +22,15 @@ def create
     @synapse = Synapse.new(synapse_params)
     @synapse.desc = '' if @synapse.desc.nil?
     @synapse.desc.strip! # no trailing/leading whitespace
-    authorize @synapse
+
+    success = nil
+    ActiveRecord::Base.transaction do
```

Comment:
Here's the issue: I was getting unathorized when I gave invalid params. I'd rather get unprocessable entity when i give invalid params. So I have to run @synapse.save, and THEN authorize. If it doesn't authorize, I want it to roll back. So the ActiveRecord transaction will run SQL commands + rails commands, and if it errors out it will roll back the transaction and never write it to disk